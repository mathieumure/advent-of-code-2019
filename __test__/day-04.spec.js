import {
    run,
    computePart1,
    computePart2,
    hasOnlyIncrementalChar,
    isCompliant,
    isCompliantWithOnlyTwoSameDigit
} from '../day-04'

describe('day-O4 - part 1', () => {

    it.each`
    password | result
    ${'123456'} | ${true}
    ${'111111'} | ${true}
    ${'223450'} | ${false}
    ${'989999'} | ${false}
    `('hasOnlyIncrementalChar', ({password, result}) => {
        expect(hasOnlyIncrementalChar(password)).toEqual(result)
    })

    it.each`
    password | result
    ${'123456'} | ${false}
    ${'111111'} | ${true}
    ${'223450'} | ${false}
    ${'989999'} | ${false}
    `('$password isCompliant', ({password, result}) => {
        expect(isCompliant(password)).toEqual(result)
    })

    it('should run', async () => {
        const result = await run('part1')
        expect(result).toMatchSnapshot()
    })
})

describe('day-O2 - part 2', () => {

    it.each`
    password | result
    ${'123456'} | ${false}
    ${'111111'} | ${false}
    ${'223450'} | ${false}
    ${'989999'} | ${false}
    ${'112233'} | ${true}
    ${'123444'} | ${false}
    ${'111122'} | ${true}
    `('$password isCompliant', ({password, result}) => {
        expect(isCompliantWithOnlyTwoSameDigit(password)).toEqual(result)
    })

    it('should run', async () => {
        const result = await run('part2')
        expect(result).toMatchSnapshot()
    })

})
