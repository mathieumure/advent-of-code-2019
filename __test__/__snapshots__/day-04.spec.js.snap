// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`day-O2 - part 2 should run 1`] = `1111`;

exports[`day-O4 - part 1 should run 1`] = `"171309-643603"`;
